xarray/core/dataarray.py

```diff
@@ -1995,11 +1996,14 @@ def rank(self, dim):
         import bottleneck as bn
         axis = self.get_axis_num(dim)
         func = bn.nanrankdata if self.dtype.kind is 'f' else bn.rankdata
-        return apply_ufunc(func, self,
-                           dask='parallelized',
-                           keep_attrs=True,
-                           output_dtypes=[np.float_],
-                           kwargs=dict(axis=axis)).transpose(*self.dims)
+        ranks = apply_ufunc(func, self,
+                            dask='parallelized',
+                            keep_attrs=True,
+                            output_dtypes=[np.float_],
+                            kwargs=dict(axis=axis)).transpose(*self.dims)
+        if not pct:
+            return ranks
+        return (ranks - 1) / (self.count(dim) - 1)
```

Comment:
@shoyer with `pct=True` this is not the same implementation as pandas which returns `rank / count`. I think this one makes more sense but users might be suprised

Reply:
Good catch, thanks for calling this out. The pandas definition makes a little more sense to me, as "fraction of the data with this rank or higher". I don't know how I would describe your version in a simple sentence.

Reply:
Normalized ? It lies in [0, 1] not [1/Nsamples, 1]

My main issue is that I would expect the pct rank to average to 0.5 but with pandas we have
```python
In [2]: pd.Series([1,2,3]).rank(pct=True)
Out[2]: 
0    0.333333
1    0.666667
2    1.000000
```

Reply:
> My main issue is that I would expect the pct rank to average to 0.5

I agree that this would be a nice property, but I'm not sure it's worth the loss in interpretability.

Reply:
How about a str arg for specifying the normalization scheme ?
'pct', 'norm' and 'polar'

Reply:
What would `polar` mean?

All things being equal, we do try to lean towards doing what pandas does in xarray because consistency makes it easier to use both packages together.

Reply:
polar would be [-1, 1]
I get your point of view, i'll make the change to mimick pandas